pages/view/_iscnId.vue

```diff
@@ -191,41 +390,80 @@ export default class ViewIscnIdPage extends Vue {
     return this.record.stakeholders
   }
 
-  created() {
-    const { iscnId } = this.$route.params
-    if (iscnId.startsWith(ISCN_PREFIX)) {
-      this.iscnId = iscnId
-    }
+  get transactionsURL() {
+    return `${BIG_DIPPER_TRANSACTIONS}${this.txHash}`
   }
 
-  async mounted() {
-    if (!this.iscnId) {
-      const param = this.$route.params.iscnId
-      if (!isCosmosTransactionHash(param)) {
-        this.$nuxt.error({
-          statusCode: 400,
-          message: ErrorMessage.statusCode400,
-        })
-        return
-      }
-      const res = await this.fetchISCNByTx(param)
-      if (!res) {
-        this.$nuxt.error({
-          statusCode: 400,
-          message: ErrorMessage.statusCode400,
-        })
-        return
-      }
-      this.iscnId = res.records[0].id
-      this.$router.replace({ params: { iscnId: this.iscnId } })
+  get rawDataURL() {
+    return `${ISCN_TX_RAW_DATA_ENDPOINT}${this.iscnId}`
+  }
+
+  async fetchTxHash() {
+    const txHash = await this.$axios.get(`${ISCN_TX_RAW_DATA_ENDPOINT}${this.iscnId}`)
+      .then((res) =>this.getTxHash(res.data.txs))
+    return txHash
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  getTxHash(items: any) {
+      const index = items.findIndex((item: any) =>
+        item.logs.find((log: any) =>
+          log.events.find((event: any) =>
+            event.attributes.some(
+              (attribute: any) =>
+                attribute.key === 'action' &&
+                attribute.value === 'create_iscn_record',
+            ),
+          ),
+        ),
+      )
+      return items[index].txhash
     }
-    if (!this.getISCNById(this.iscnId) || !this.owner) {
-      const res = await this.fetchISCNById(this.iscnId)
-      if (res) this.owner = res.owner
+
+  showStakeholder(index: number) {
+    this.isOpenAuthorDialog = true
+    const {
+      id,
+      likerId,
+      description: authorDescription,
+      walletAddress: authorWalletAddress,
+      url: authorUrl,
+      name: authorName,
+    } = this.stakeholders[index].entity
+    this.stakeholderInfo = {
+      id,
+      likerId,
+      authorDescription,
+      authorWalletAddress,
+      authorUrl,
+      authorName,
     }
-    if (!this.getISCNById(this.iscnId)) {
-      this.$nuxt.error({ statusCode: 404, message: ErrorMessage.statusCode404 })
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  handleCopy(address: string, type: string) {
+    let text = ''
+    if (type === 'cosmos') {
+      text = address.replace(/(did:|:)/g, '')
+    } else {
+      text = address.replace(/(did:|eth:|:)/g, '')
```

Comment:
If `cosmos` address is the only special case, I think we can just remove anything `did:*:` instead of hardcoding eth

Reply:
seems we are still hardcoding `eth` here?